Topics/System_Design/behavioral_design_patterns.md

```diff
@@ -0,0 +1,120 @@
+# Behavioral Design Patterns
+
+In software engineering, behavioral design patterns  are a set of patterns that focus on the interaction between objects and how they communicate and collaborate with each other. These patterns address the assignment of responsibilities between objects, encapsulation of behavior, and the flow of control among them. They aim to improve the flexibility, extensibility, and maintainability of software systems by providing solutions to common communication and collaboration problems.
+
+## Table of Contents
+
+1. [Usefulness of Behavioral Design Patterns](#use)
+2. Examples of Behavioral Design Patterns
+    1. [Observer](#obs)
+    2. [Strategy](#str)
+    3. [Iterator](#itr)
+    4. [Command](#com)
+    5. Examples of other [Behavioral Design Patterns](https://refactoring.guru/design-patterns/behavioral-patterns)
```

Comment:
```suggestion
    5. [Examples of other Behavioral Design Patterns](https://refactoring.guru/design-patterns/behavioral-patterns)
```
I think it would make more sense if the entire note was the link

Reply:
Good catch! I agree!